Sources/Widgets/Core/WidgetManager/index.js

```diff
@@ -288,38 +298,18 @@ function vtkWidgetManager(publicAPI, model) {
     }
   }
 
-  const handleEvent = (eventName) => {
-    let guard = false;
-    return (callData) => {
-      if (
-        guard ||
-        model.isAnimating ||
-        !model.pickingEnabled ||
-        model._selectionInProgress
-      ) {
-        return macro.VOID;
-      }
-
-      const updatePromise = updateSelection(callData);
-      macro.measurePromiseExecution(updatePromise, (elapsed) => {
-        // 100ms is deemed fast enough. Anything higher can degrade usability.
-        if (elapsed > 100) {
-          macro.vtkWarningMacro(
-            `vtkWidgetManager updateSelection() took ${elapsed}ms on ${eventName}`
-          );
-        }
-      });
+  const handleEvent = async (callData, fromTouchEvent = false) => {
```

Comment:
this function should always return something

Reply:
handleEvent doesn't need to return anything, it has nothing to return. The event handling logic treats "undefined" returns the same as "macro.VOID".

Reply:
Even if undefined (nothing returned) is the same as "macro.VOID", I find it a good practice to have event callbacks return something. This gives some hint to the developer reading /modifying/copying your code.

I've seen many cases where nothing was returned but an ABORT should have been returned. And when I asked the developer, they typically said "I didn't know that callback could return something".

Reply:
Ah, well in that case I'm good with returning macro.VOID explicitly for clarity purposes.